### ISSUE TITLE
Subsample PRG "founder" variants

### ISSUE THREAD

**Issue by mbhall88** [mbhall88/head_to_head_pipeline#14]
[continuation of #2]  
We will no longer use FORGe to select variants to use for our PRG construction [#3]. Instead, we will randomly subsample from lineage-specific VCFs [#11] for lineages with >100 samples. If there are less than 100 samples in a VCF, we include *all* variants from that VCF.

**Comment by mbhall88** [comment]
The main decision for this step is what number of samples to use for the sparse and dense PRGs @iqbal-lab 

**Initial suggestion**:  
- Sparse: Max 100 samples from each lineage
- Dense: Max 500 samples from each lineage

**Comment by iqbal-lab** [comment]
I think this needs just a small bit of modelling, and basically will just need to try some values and see what the effect is. Here's the modelling, feel free to code this up and explore a bit

If a variant/haplotype is at frequency f, and i pick a genome for the PRG, i get that variant/haplotype with probability=f. If i pick N genomes, I get that var/hap with probability = 1 - (1-f)^N. This is a standard trick - basically it's 1- the probability of missing it each of the N times.

So, if I pick 100 genomes, the probability of getting haplotypes at different frequencies is given by 1 - (1-f)^100.

In our case, for the proposed sparse number above, we can just plot this and see what we think
In R
`prob<-function(freq)
{ 

return (1 - (1-freq)^100)

}
f<-c(0.0001, 0.001, 0.005, 0.01,0.02, 0.03, 0.04, 0.05, 0.5)
plot(f,prob(f), type="b", main="What frequencies do we catch with 100 genomes")
` 

![results100](https://user-images.githubusercontent.com/6476513/84241005-c1e66500-aaf6-11ea-98e7-289699a63564.png)

So you can see, with that sparse choice, we expect to get everything above 5% frequency.

> prob(0.05)
[1] 0.9940795

Seems good to me.

What about the dense option - picking 500

![results500](https://user-images.githubusercontent.com/6476513/84241319-3a4d2600-aaf7-11ea-9c2a-461b612236b2.png)

basically,this takes you down to getting stuff with freq >= 2%

I think those suggestions are fine. We'll need to build, get results, and see what the cost/benefit is of bigger PRGs

**Comment by mbhall88** [comment]
Ok, repeating the above analysis but printing the raw values:

```py
def prob(freq: float, n: int) -> float:
    return 1 - (1 - freq)**n
```
### 5% frequency

```py
>>> freq = 0.05
>>> for n in [50, 100, 200, 500, 1000, 2000]:
>>>     print(n, prob(freq, n))
50 0.9230550247232868
100 0.994079470779666
200 0.9999649473337512
500 0.9999999999927255
1000 1.0
2000 1.0
```

### 1% frequency

```py
>>> freq = 0.01
>>> for n in [50, 100, 200, 500, 1000, 2000]:
>>>     print(n, prob(freq, n))
50 0.39499393286246365
100 0.6339676587267709
200 0.8660203251420383
500 0.9934295169575854
1000 0.9999568287525893
2000 0.9999999981362434
```

So, as you say, 100 and 500 seem to be very sane starting points

**Comment by mbhall88** [comment]
Single VCFs now exist for each of the subsampled frequencies above.  
The following filters were applied to the VCF to trim it down to positions of interest:
- Only keep positions where at least one sample has an ALT call
- Only keep positions not in the compass mask
- Only keep positions where `FILTER=.|PASS`

After applying these filters, the VCF files contain:

| type | # variants | filepath |
| --- | --- | --- |
| sparse | 55,173 | `/hps/nobackup/research/zi/projects/tech_wars/data/H37Rv_PRG/vcfs/filtered/subsampled-100.filtered.bcf.gz` |
| dense | 160,003 | `/hps/nobackup/research/zi/projects/tech_wars/data/H37Rv_PRG/vcfs/filtered/subsampled-500.filtered.bcf.gz` |
